extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java

```diff
@@ -0,0 +1,251 @@
+            List<WorkspaceActionBuildItem> workspaceActionBuildItems,
+            BuildProducer<BuildTimeActionBuildItem> buildTimeActionProducer) {
+
+        // Dev UI Build Time Actions
+        BuildTimeActionBuildItem buildItemActions = new BuildTimeActionBuildItem(NAMESPACE);
+
+        // Workspace Actions
+        Map<String, Action> actionMap = workspaceActionBuildItems.stream()
+                .flatMap(item -> item.getActions().stream())
+                .map(ActionBuilder::build)
+                .collect(Collectors.toMap(Action::getId, action -> action, (a, b) -> a));
+
+        buildItemActions.addAction("getWorkspaceItems", (t) -> {
+            return workspaceBuildItem.getWorkspaceItems();
+        });
+
+        buildItemActions.addAction("getWorkspaceActions", (t) -> {
+            return actionMap.values().stream()
+                    .map(action -> new WorkspaceAction(action.getId(), action.getLabel(), action.getFilter(),
+                            action.getDisplay(), action.getDisplayType()))
+                    .sorted(Comparator.comparing(WorkspaceAction::label))
+                    .collect(Collectors.toList());
+        });
+
+        buildItemActions.addAction("executeAction", (t) -> {
+
+            String actionId = t.get("actionId");
+
+            if (actionId != null) {
+                Path path = Path.of(URI.create(t.get("path")));
+                Action actionToExecute = actionMap.get(actionId);
+                Path convertedPath = (Path) actionToExecute.getPathConverter().apply(path);
+                return new WorkspaceActionResult(convertedPath, actionToExecute.getFunction().apply(t));
+            }
+            return null;
+        });
+
+        buildItemActions.addAction("getWorkspaceItemContent", (Map<String, String> params) -> {
+            if (params.containsKey("path")) {
+                Path path = Paths.get(URI.create(params.get("path")));
+                return readContents(path);
+            }
+            return null;
+        });
+
+        buildItemActions.addAction("saveWorkspaceItemContent", (Map<String, String> params) -> {
+            if (params.containsKey("content")) {
+                String content = params.get("content");
+                Path path = Paths.get(URI.create(params.get("path")));
+                writeContent(path, content);
+                return new SavedResult(path, true, null);
+            }
+            return new SavedResult(null, false, "Invalid input");
+        });
+
+        buildTimeActionProducer.produce(buildItemActions);
+    }
+
+    private void sortWorkspaceItems(List<WorkspaceBuildItem.WorkspaceItem> items) {
+        items.sort(Comparator.comparing((WorkspaceBuildItem.WorkspaceItem item) -> isFileInRoot(item.name()))
+                .thenComparing(item -> folderPriority(item.name()))
+                .thenComparing(WorkspaceBuildItem.WorkspaceItem::name));
+    }
+
+    private static int folderPriority(String name) {
```

Comment:
This is very Maven specific; we need to be more flexible.

Reply:
This will work for maven and gradle.

Reply:
My concern is more JBang.

Reply:
The sources at not structured conventionally.

Reply:
Is this a show stopper ? Again, all that this means is that the order in the ui will not be /src/main , /src/test etc. But it will still work.

Reply:
i haven't tried it yet but i dont think its a problem for jbang - since jbang's structure/projects are less simpler it doesn't get as crowded to need special treatment :)

I would maybe just also included src/ as a last high priority.

Reply:
Done, as per Max's suggestion, I added /src - Marking this resolved.